k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1489,12 +1533,28 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             }
         } else {
             currentMessageBuilder = null;
+
+            if (mAction == Action.REDIRECT) {
```

Comment:
It doesn't matter what should or shouldn't be done to the message, there should be no logic that changes (or creates) the message here. that's the MessageBuilder's job